src/overlays/actors/ovl_En_Zf/z_en_zf.c

```diff
@@ -82,115 +151,2291 @@ static ColliderQuadInit D_80B4A204 = {
+}
+
+void EnZf_JumpForward(EnZf* this, GlobalContext* globalCtx) {
+    if ((this->unk_3F0 != 0) && (this->actor.world.pos.y <= this->actor.floorHeight)) {
+        this->actor.world.pos.y = this->actor.floorHeight;
+        this->hopAnimIndex = 0;
+        this->actor.velocity.y = 0.0f;
+        this->actor.speedXZ = 0.0f;
+    }
+
+    if (SkelAnime_Update(&this->skelAnime)) {
+        if (this->unk_3F0 == 0) {
+            Animation_Change(&this->skelAnime, &gZfLandingAnim, 3.0f, 0.0f, 17.0f, 2, -3.0f);
+            this->unk_3F0 = 10;
+            Audio_PlayActorSound2(&this->actor, NA_SE_EN_RIZA_JUMP);
+        } else {
+            this->actor.speedXZ = 0.0f;
+            this->hopAnimIndex = 0;
+            EnZf_SetupApproachPlayer(this, globalCtx);
+        }
+    }
+    if ((globalCtx->gameplayFrames & 0x5F) == 0) {
+        Audio_PlayActorSound2(&this->actor, NA_SE_EN_RIZA_CRY);
+    }
+
+    if ((this->actor.params == ENZF_TYPE_DINOLFOS) && (this->actor.bgCheckFlags & 3)) {
+        if (EnZf_MayAttack(globalCtx, this)) {
+            EnZf_SetupSlash(this);
+        } else {
+            func_80B483E4(this, globalCtx);
+        }
+    }
+}
+
+void func_80B4604C(EnZf* this) {
+    Animation_MorphToLoop(&this->skelAnime, &gZfWalkingAnim, -4.0f);
+    this->action = ENZF_ACTION_6;
+    EnZf_SetupAction(this, func_80B46098);
+}
+
+void func_80B46098(EnZf* this, GlobalContext* globalCtx) {
+    s32 pad;
+    f32 phi_f2;
+    Player* player = GET_PLAYER(globalCtx);
+    s16 temp_v0;
+    s16 phi_v1;
+
+    if (!EnZf_DodgeRangedEngaging(globalCtx, this)) {
+        if ((this->actor.params != ENZF_TYPE_DINOLFOS) || !EnZf_ChooseAction(globalCtx, this)) {
+            temp_v0 = this->actor.yawTowardsPlayer - this->actor.shape.rot.y;
+
+            if (temp_v0 > 0) {
+                phi_v1 = (temp_v0 * 0.25f) + 2000.0f;
+            } else {
+                phi_v1 = (temp_v0 * 0.25f) - 2000.0f;
+            }
+
+            this->actor.shape.rot.y += phi_v1;
+            this->actor.world.rot.y = this->actor.shape.rot.y;
+
+            if (temp_v0 > 0) {
+                phi_f2 = phi_v1 * 1.5f;
+                if (phi_f2 > 2.0f) {
+                    phi_f2 = 2.0f;
+                }
+            } else {
+                phi_f2 = phi_v1 * 1.5f;
+                if (phi_f2 < -2.0f) {
+                    phi_f2 = -2.0f;
+                }
+            }
+
+            this->skelAnime.playSpeed = -phi_f2;
+            SkelAnime_Update(&this->skelAnime);
+
+            // miniboss
+            if (this->actor.params >= ENZF_TYPE_LIZALFOS_MINIBOSS_A) {
+                this->curPlatform = EnZf_FindPlatform(&this->actor.world.pos, this->curPlatform);
+                if (this->curPlatform != EnZf_FindPlatform(&player->actor.world.pos, -1)) {
+                    EnZf_SetupApproachPlayer(this, globalCtx);
+                    return;
+                }
+            }
+
+            if (Actor_IsFacingPlayer(&this->actor, 30 * 0x10000 / 360)) {
```

Comment:
```suggestion
            if (Actor_IsFacingPlayer(&this->actor, DEGF_TO_BINANG(30))) {
```

Reply:
I don't think this will work since it's not a float calculation.